src/run_spineopt.jl

```diff
@@ -171,6 +172,56 @@ function run_spineopt(
     end
 end
 
+function run_spineopt(
+    url_in::Dict,
+    url_out::Union{String,Nothing}=nothing;
+    upgrade=false,
+    mip_solver=nothing,
+    lp_solver=nothing,
+    add_user_variables=m -> nothing,
+    add_constraints=m -> nothing,
+    log_level=3,
+    optimize=true,
+    update_names=false,
+    alternative="",
+    write_as_roll=0,
+    use_direct_model=false,
+    filters=Dict("tool" => "object_activity_control"),
+    templates=(),
+    log_file_path=nothing,
+    resume_file_path=nothing,
+    run_kernel=run_spineopt_kernel!,
+)
+    db_url = "sqlite://"
+    data_template = Dict(Symbol(key) => value for (key, value) in template())#SpineOpt.template()
+    data_in = Dict(Symbol(key) => value for (key, value) in url_in)
+    data_run = merge(data_template, data_in)
+    SpineInterface.close_connection(db_url)
+    SpineInterface.open_connection(db_url)
+    SpineInterface.import_data(db_url, "From Julia dictionary to Spine database."; data_run...)
```

Comment:
This requires PyCall to be properly configured, are we sure we want that?

`using_spinedb` already accepts a dictionary as an input. If you look at `prepare_spineopt`, you will see that we actually use that method. We call `data = export_data(url_in; upgrade=upgrade, filters=filters)` followed by `using_spinedb(data, @__MODULE__; extend=true)`. But if `url_in` is already a dict, then we could just replace the former with `data = url_in` and proceed.

Reply:
We don't need to define an entire new `run_spineopt` method. Maybe we should try that idea?

Reply:
I'm not really sure whether I fully understand all of this, but I'm certainly willing to try.

But can we really avoid the new run_spineopt? Do you want a check whether the input is a string or a dictionary and then run the one or the other? And would run_spineopt then take something like run_spineopt(url_in::Union{String, Dict})?

Reply:
Sounds good to me